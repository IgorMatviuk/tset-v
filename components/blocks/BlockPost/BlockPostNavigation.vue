<template>
  <div class="post-navigation">
    <BContainer>
      <hr class="post-navigation__delimiter" />
      <div class="post-navigation__content">
        <NuxtLink
          v-if="links.prev"
          :to="$i18n.path(`blog/${links.prev.slug}`)"
          class="post-navigation__link post-navigation__link--prev"
        >
          <svg-icon
            name="arrow"
            class="post-navigation__icon post-navigation__icon--prev"
            width="30"
            height="9"
          />
          <span class="post-navigation__link-text">
            {{ links.prev.title }}
          </span>
        </NuxtLink>
        <NuxtLink
          v-if="links.next"
          :to="$i18n.path(`blog/${links.next.slug}`)"
          class="post-navigation__link post-navigation__link--next"
        >
          <span class="post-navigation__link-text">
            {{ links.next.title }}
          </span>
          <svg-icon
            name="arrow"
            class="post-navigation__icon post-navigation__icon--next"
            width="30"
            height="9"
          />
        </NuxtLink>
      </div>
    </BContainer>
  </div>
</template>

<script>
export default {
  props: {
    links: {
      type: Object,
      required: true
    }
  }
}
</script>

<style lang="scss">
.post-navigation {
  &__delimiter {
    margin-top: 12px;
    margin-bottom: 19px;
    border-top: 2px solid $primary-light-2;
  }

  &__content {
    display: flex;
    align-items: center;
    justify-content: space-between;
  }

  &__link {
    display: flex;
    align-items: center;
    transition: color 0.3s;

    &--prev {
      margin-right: auto;
    }

    &--next {
      margin-left: auto;
    }
  }

  &__link-text {
    display: inline-block;
    max-width: 80px;
    overflow: hidden;
    font-size: 14px;
    font-weight: 300;
    line-height: 18px;
    text-overflow: ellipsis;
    text-transform: uppercase;
    white-space: nowrap;
  }

  &__icon {
    width: 30px;
    height: 9px;
    margin-bottom: 2px;

    &--prev {
      margin-right: 17px;
      transform: rotate(180deg);
    }

    &--next {
      margin-left: 5px;
    }
  }

  @include media-breakpoint-up(md) {
    &__delimiter {
      margin-bottom: 13px;
    }

    &__content,
    &__delimiter {
      width: 450px;
      margin-right: auto;
      margin-left: auto;
    }

    &__link-text {
      max-width: 180px;
      font-size: 11px;
      line-height: 14px;
    }

    &__icon {
      &--prev {
        margin-right: 5px;
      }

      &--next {
        margin-left: 1px;
      }
    }
  }

  @include media-breakpoint-up(lg) {
    &__delimiter {
      margin-bottom: 15px;
    }

    &__content,
    &__delimiter {
      width: 610px;
    }

    &__link-text {
      max-width: 255px;
    }
  }

  @include media-breakpoint-up(xl) {
    &__delimiter {
      margin-bottom: 25px;
    }

    &__content,
    &__delimiter {
      width: 770px;
    }

    &__link-text {
      max-width: 320px;
      font-size: 14px;
      line-height: 18px;
    }

    &__icon {
      &--prev {
        margin-right: 13px;
      }
    }
  }

  @include media-breakpoint-up(xxl) {
    &__delimiter {
      margin-bottom: 24px;
    }

    &__content,
    &__delimiter {
      width: 930px;
    }

    &__link-text {
      max-width: 365px;
      font-size: 16px;
      line-height: 20px;
    }

    &__icon {
      &--prev {
        margin-right: 20px;
      }

      &--next {
        margin-left: 10px;
      }
    }
  }
}
</style>
