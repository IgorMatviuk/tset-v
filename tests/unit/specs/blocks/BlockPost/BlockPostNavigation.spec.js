import { mount, createLocalVue } from '@vue/test-utils'

import { BContainer } from 'bootstrap-vue'

import BlockPostNavigation from '@/components/blocks/BlockPost/BlockPostNavigation'

const localVue = createLocalVue()
localVue.component('BContainer', BContainer)

describe('BlockPostNavigation', () => {
  function createBlockPostNavigation(props) {
    const wrapper = mount(BlockPostNavigation, {
      localVue,
      propsData: {
        ...props
      },
      mocks: {
        $i18n: {
          path: (url) => url
        }
      },
      stubs: ['svg-icon', 'NuxtLink']
    })

    return wrapper
  }

  test('render post navigation', () => {
    const links = {
      prev: {
        slug:
          'sovet-v-dekret-kak-podobrat-stilnye-letnie-platya-i-sarafany-dlya-beremennyh',
        title:
          'Совет в декрет! Как подобрать стильные летние платья и сарафаны для беременных?'
      },
      next: {
        slug:
          'made-in-ukraine-massmarket-protiv-ukrainskogo-brenda-chto-vybrat',
        title:
          'Made in Ukraine! Массмаркет против украинского бренда, что выбрать?'
      }
    }

    const wrapper = createBlockPostNavigation({ links })

    expect(wrapper.element).toMatchSnapshot()
  })

  test('render post navigation without next link', () => {
    const links = {
      prev: {
        slug:
          'sovet-v-dekret-kak-podobrat-stilnye-letnie-platya-i-sarafany-dlya-beremennyh',
        title:
          'Совет в декрет! Как подобрать стильные летние платья и сарафаны для беременных?'
      }
    }

    const wrapper = createBlockPostNavigation({ links })

    expect(wrapper.element).toMatchSnapshot()
  })

  test('render post navigation without previous link', () => {
    const links = {
      next: {
        slug:
          'sovet-v-dekret-kak-podobrat-stilnye-letnie-platya-i-sarafany-dlya-beremennyh',
        title:
          'Совет в декрет! Как подобрать стильные летние платья и сарафаны для беременных?'
      }
    }

    const wrapper = createBlockPostNavigation({ links })

    expect(wrapper.element).toMatchSnapshot()
  })
})
