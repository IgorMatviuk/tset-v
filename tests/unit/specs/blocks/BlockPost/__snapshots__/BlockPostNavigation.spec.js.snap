// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockPostNavigation render post navigation 1`] = `
<div
  class="post-navigation"
>
  <div
    class="container"
  >
    <hr
      class="post-navigation__delimiter"
    />
     
    <div
      class="post-navigation__content"
    >
      <nuxtlink-stub
        class="post-navigation__link post-navigation__link--prev"
        to="blog/sovet-v-dekret-kak-podobrat-stilnye-letnie-platya-i-sarafany-dlya-beremennyh"
      >
        <svg-icon-stub
          class="post-navigation__icon post-navigation__icon--prev"
          height="9"
          name="arrow"
          width="30"
        />
         
        <span
          class="post-navigation__link-text"
        >
          
          Совет в декрет! Как подобрать стильные летние платья и сарафаны для беременных?
        
        </span>
      </nuxtlink-stub>
       
      <nuxtlink-stub
        class="post-navigation__link post-navigation__link--next"
        to="blog/made-in-ukraine-massmarket-protiv-ukrainskogo-brenda-chto-vybrat"
      >
        <span
          class="post-navigation__link-text"
        >
          
          Made in Ukraine! Массмаркет против украинского бренда, что выбрать?
        
        </span>
         
        <svg-icon-stub
          class="post-navigation__icon post-navigation__icon--next"
          height="9"
          name="arrow"
          width="30"
        />
      </nuxtlink-stub>
    </div>
  </div>
</div>
`;

exports[`BlockPostNavigation render post navigation without next link 1`] = `
<div
  class="post-navigation"
>
  <div
    class="container"
  >
    <hr
      class="post-navigation__delimiter"
    />
     
    <div
      class="post-navigation__content"
    >
      <nuxtlink-stub
        class="post-navigation__link post-navigation__link--prev"
        to="blog/sovet-v-dekret-kak-podobrat-stilnye-letnie-platya-i-sarafany-dlya-beremennyh"
      >
        <svg-icon-stub
          class="post-navigation__icon post-navigation__icon--prev"
          height="9"
          name="arrow"
          width="30"
        />
         
        <span
          class="post-navigation__link-text"
        >
          
          Совет в декрет! Как подобрать стильные летние платья и сарафаны для беременных?
        
        </span>
      </nuxtlink-stub>
       
      <!---->
    </div>
  </div>
</div>
`;

exports[`BlockPostNavigation render post navigation without previous link 1`] = `
<div
  class="post-navigation"
>
  <div
    class="container"
  >
    <hr
      class="post-navigation__delimiter"
    />
     
    <div
      class="post-navigation__content"
    >
      <!---->
       
      <nuxtlink-stub
        class="post-navigation__link post-navigation__link--next"
        to="blog/sovet-v-dekret-kak-podobrat-stilnye-letnie-platya-i-sarafany-dlya-beremennyh"
      >
        <span
          class="post-navigation__link-text"
        >
          
          Совет в декрет! Как подобрать стильные летние платья и сарафаны для беременных?
        
        </span>
         
        <svg-icon-stub
          class="post-navigation__icon post-navigation__icon--next"
          height="9"
          name="arrow"
          width="30"
        />
      </nuxtlink-stub>
    </div>
  </div>
</div>
`;
